scraper/free_scraper/pipelines.py

```diff
@@ -80,7 +87,7 @@ class ConsoleLogPipeline(object):
     """
     def process_item(self, item, spider):
         print "----------------FreeShittyEyeOut by /u/leesw----------------"
-        print "Course: ", item['course']['name'] + item['section']
+        print "Course: ", item.get("course").get("name") + item.get("section")
         print "Total Seats Remaining: ", item['seats_data']['total_seats']
```

Comment:
if we're gonna `.get()` then might as well go all the way and `.get()` everything

Reply:
`.get()` deals with NULL and gives 0/NULL if nothing exists there.
`[]` would give empty string.

Reply:
In that case, isn't `[]' ideal, since we are printing strings?

Reply:
@leesw98 Haha yeah I was just writing up a reply about that

The advantage to `get()` is you can specify a default value if key is not found:
```py
dict.get('key', 'default value') # return 'default value' if key not found
```
though it does return `None` by default. It might be a good idea to specify a default in this case, ie:
```py
print 'Course: ', item.get('course').get('name', 'N/A') # print "Course: N/A" if name not found
```
**However, all this is pretty useless because we never return `None`**: look at the [safe extract helper](https://github.com/leesw98/FreeEyeOut/blob/413b7143b0d01f6fe06556e8146d57e885da6bae/scraper/free_scraper/spiders/parsers/course_parser.py#L120) - we only return strings, and never `None`.

Interesting tibbit: the reason `dic['key']` was/kinda still is the standard option is because on older versions of Python, `.get()` was significantly slower

Reply:
oh just noticed you reverted these changes in 7ee7799. nice 👍